packages/file_selector/file_selector_ios/ios/Classes/FTLFileSelectorPlugin.m

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FTLFileSelectorPlugin.h"
+
+@interface FTLFileSelectorPlugin () <UIDocumentPickerDelegate>
+
+@property(nonatomic) FlutterResult pendingResult;
+
+@end
+
+@implementation FTLFileSelectorPlugin
+
+- (void)openPickerWithCall:(FlutterMethodCall *)call
+                    result:(FlutterResult)result
+       allowMultiSelection:(BOOL)allowMultiSelection {
+  NSDictionary *acceptedTypes = call.arguments[@"acceptedTypes"];
+  UIDocumentPickerViewController *documentPicker =
+      [[UIDocumentPickerViewController alloc] initWithDocumentTypes:acceptedTypes[@"UTIs"]
+                                                             inMode:UIDocumentPickerModeImport];
+  documentPicker.delegate = self;
+  if (@available(iOS 11.0, *)) {
+    documentPicker.allowsMultipleSelection = allowMultiSelection;
+  }
+
+  UIViewController *rootVC = UIApplication.sharedApplication.delegate.window.rootViewController;
+  if (rootVC) {
+    [rootVC presentViewController:documentPicker animated:YES completion:nil];
+    self.pendingResult = result;
```

Comment:
Because this is globally stateful, you'll silently fail to ever reply to the first call if someone calls `open*` twice in a row. You should check for `pendingResult` being non-nil before doing anything else, and return an error if it is.